artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/server/impl/RemotingServiceImpl.java

```diff
@@ -554,20 +558,13 @@ public void connectionDestroyed(final Object connectionID) {
                return;
             }
          }
-         conn.connection.fail(new ActiveMQRemoteDisconnectException());
+         conn.connection.fail(e);
       }
    }
 
    @Override
    public void connectionException(final Object connectionID, final ActiveMQException me) {
-      // We DO NOT call fail on connection exception, otherwise in event of real connection failure, the
-      // connection will be failed, the session will be closed and won't be able to reconnect
-
-      // E.g. if live server fails, then this handler wil be called on backup server for the server
-      // side replicating connection.
-      // If the connection fail() is called then the sessions on the backup will get closed.
-
-      // Connections should only fail when TTL is exceeded
+      issueFailure(connectionID, me);
    }
```

Comment:
Does this comment no longer apply?  Is there not a risk that core clients won't be able to reconnect to their server session in the case of a spurious network failure?

Reply:
no.. it didn't apply for some time already.

This is legacy from hornetq where we would only issue disconnects based on TTL.

Reply:
I'm running the full testsuite to make sure.